src/main/java/pl/javastart/task/DataReader.java

```diff
@@ -0,0 +1,130 @@
+package pl.javastart.task;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.Scanner;
+
+public class DataReader {
+    private static final int EXIT = 0;
+    private static final int ADD_NEW = 1;
+    private static final int  GET_NEXT = 2;
+
+    Scanner scanner = new Scanner(System.in);
+
+    private Queue<Vehicle> vehicles = new LinkedList<>();
+
+    public void chooseOption() {
+        int option;
+        do {
+            printOptions();
+            option = scanner.nextInt();
+            scanner.nextLine();
+            switch (option) {
+                case EXIT:
+                    exit();
+                    break;
+                case ADD_NEW:
+                    addNew();
+                    break;
+                case GET_NEXT:
+                    getNext();
+                    break;
+                default:
+                    System.out.println("Wybrana opcja nie istnieje. Podaj jeszcze raz.");
+            }
+        } while (option != EXIT);
+    }
+
+    public void close() {
+        scanner.close();
+    }
+
+    private void getNext() {
+        Vehicle vehicle = vehicles.poll();
+        System.out.println("Nastepny pojazd do kontroli:");
+        System.out.println(vehicle);
+    }
+
+    private void exit() {
+        writeVehiclesToFile();
+        System.out.println("Koniec programu");
+    }
+
+    private void writeVehiclesToFile() {
+        String fileName = "vehicles.csv";
+        File file = new File(fileName);
+        try {
+            FileWriter writer = new FileWriter(file);
+            for (Vehicle vehicle : vehicles) {
+                writer.write(String.valueOf(vehicle));
+            }
+            writer.flush();
+        } catch (IOException e) {
+            System.err.println("Nie udalo sie zapisac danych do pliku");
+        }
+    }
+
+    private void addNew() {
+        Vehicle vehicle = insertVehicleData();
+        vehicles.offer(vehicle);
+
+    }
+
+    private Queue<Vehicle> importVehiclesFromFile() {
+        String fileName = "vehicles.csv";
+        File file = new File(fileName);
```

Comment:
Pamiętaj o zamykaniu pliku, albo najlepiej wprowadź try-with-resources https://youtu.be/LRwSHA9YDgY